### PR TITLE
Replace asserts with raising a ValidatorError

### LINKED ISSUE #64: Replace assert statements from source
Running [bandit](https://bandit.readthedocs.io/) on our repository revealed the following security issues:
```
>> Issue: [B101:assert_used] Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
   Severity: Low   Confidence: High
   CWE: CWE-703 (https://cwe.mitre.org/data/definitions/703.html)
   More Info: https://bandit.readthedocs.io/en/1.7.8/plugins/b101_assert_used.html
   Location: ethstaker_deposit/key_handling/key_derivation/mnemonic.py:130:16
129	                """
130	                assert reconstructed_mnemonic is None
131	                reconstructed_mnemonic = ' '.join([_index_to_word(full_word_list, index) for index in word_indices])

--------------------------------------------------
>> Issue: [B101:assert_used] Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
   Severity: Low   Confidence: High
   CWE: CWE-703 (https://cwe.mitre.org/data/definitions/703.html)
   More Info: https://bandit.readthedocs.io/en/1.7.8/plugins/b101_assert_used.html
   Location: ethstaker_deposit/utils/intl.py:31:8
30	        ans = reduce(dict.get, mapList, dataDict)
31	        assert isinstance(ans, str)
32	        return ans

--------------------------------------------------
>> Issue: [B101:assert_used] Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
   Severity: Low   Confidence: High
   CWE: CWE-703 (https://cwe.mitre.org/data/definitions/703.html)
   More Info: https://bandit.readthedocs.io/en/1.7.8/plugins/b101_assert_used.html
   Location: ethstaker_deposit/utils/validation.py:138:8
137	        num_int = int(num)  # Try cast to int
138	        assert num_int == float(num)  # Check num is not float
139	        assert low <= num_int < high  # Check num in range

--------------------------------------------------
>> Issue: [B101:assert_used] Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
   Severity: Low   Confidence: High
   CWE: CWE-703 (https://cwe.mitre.org/data/definitions/703.html)
   More Info: https://bandit.readthedocs.io/en/1.7.8/plugins/b101_assert_used.html
   Location: ethstaker_deposit/utils/validation.py:139:8
138	        assert num_int == float(num)  # Check num is not float
139	        assert low <= num_int < high  # Check num in range
140	        return num_int

--------------------------------------------------
>> Issue: [B101:assert_used] Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
   Severity: Low   Confidence: High
   CWE: CWE-703 (https://cwe.mitre.org/data/definitions/703.html)
   More Info: https://bandit.readthedocs.io/en/1.7.8/plugins/b101_assert_used.html
   Location: ethstaker_deposit/utils/validation.py:266:8
265	    try:
266	        assert len(bls_withdrawal_credentials_bytes) == 32
267	        assert bls_withdrawal_credentials_bytes[:1] == BLS_WITHDRAWAL_PREFIX

--------------------------------------------------
>> Issue: [B101:assert_used] Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
   Severity: Low   Confidence: High
   CWE: CWE-703 (https://cwe.mitre.org/data/definitions/703.html)
   More Info: https://bandit.readthedocs.io/en/1.7.8/plugins/b101_assert_used.html
   Location: ethstaker_deposit/utils/validation.py:267:8
266	        assert len(bls_withdrawal_credentials_bytes) == 32
267	        assert bls_withdrawal_credentials_bytes[:1] == BLS_WITHDRAWAL_PREFIX
268	    except (ValueError, AssertionError):

--------------------------------------------------

Code scanned:
	Total lines of code: 2350
	Total lines skipped (#nosec): 0
	Total potential issues skipped due to specifically being disabled (e.g., #nosec BXXX): 0

Run metrics:
	Total issues (by severity):
		Undefined: 0
		Low: 6
		Medium: 0
		High: 0
	Total issues (by confidence):
		Undefined: 0
		Low: 0
		Medium: 0
		High: 6
Files skipped (0):
```

### PR DESCRIPTION
**Pull request by remyroy**
Fixes #64 and improves the results from running [bandit](https://bandit.readthedocs.io/)

### CODE REVIEW

File: ethstaker_deposit/key_handling/key_derivation/mnemonic.py [hunk 1]
```diff
@@ -127,7 +128,8 @@ def reconstruct_mnemonic(mnemonic: str, words_path: str) -> Optional[str]:
                 This check guarantees that only one language has a valid mnemonic.
                 It is needed to ensure abbrivated words aren't valid in multiple languages
                 """
-                assert reconstructed_mnemonic is None
+                if reconstructed_mnemonic is not None:
+                    raise ValidationError("This mnemonic abbrivated form is available in multiple languages.")
```
Comment (valefar-on-discord) [inline ethstaker_deposit/key_handling/key_derivation/mnemonic.py:132]: Very minor but
```suggestion
                    raise ValidationError("This mnemonic abbreviated form is available in multiple languages.")
```

Looks like the comment was initially wrong as well

File: ethstaker_deposit/utils/validation.py [hunk 2]
```diff
@@ -263,9 +265,11 @@ def validate_bls_withdrawal_credentials(bls_withdrawal_credentials: str) -> byte
         raise ValidationError(load_text(['err_is_already_eth1_form']) + '\n')
 
     try:
-        assert len(bls_withdrawal_credentials_bytes) == 32
-        assert bls_withdrawal_credentials_bytes[:1] == BLS_WITHDRAWAL_PREFIX
-    except (ValueError, AssertionError):
+        if len(bls_withdrawal_credentials_bytes) != 32:
+            raise ValidationError('Lenght is not 32')
```
Comment (valefar-on-discord) [inline ethstaker_deposit/utils/validation.py:269]: 😄 
```suggestion
            raise ValidationError('Length is not 32')
```

**Review by valefar-on-discord — APPROVED** [review]
Looks good besides a couple of spelling errors

**Comment by remyroy** [comment]
Thanks for catching those mistakes.

**Pull request merged**